src/Mapper/ORM/DoctrineORMMapper.php

```diff
@@ -0,0 +1,292 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\Doctrine\Mapper\ORM;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Event\LoadClassMetadataEventArgs;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use ReflectionException;
+use RuntimeException;
+
+final class DoctrineORMMapper implements EventSubscriber
+{
+    /**
+     * @var array
+     */
+    private $associations;
+
+    /**
+     * @var array
+     */
+    private $discriminators;
+
+    /**
+     * @var array
+     */
+    private $discriminatorColumns;
+
+    /**
+     * @var array
+     */
+    private $inheritanceTypes;
+
+    /**
+     * @var array
+     */
+    private $indexes;
+
+    /**
+     * @var array
+     */
+    private $uniques;
+
+    /**
+     * @var array
+     */
+    private $overrides;
+
+    public function __construct(array $associations = [], array $indexes = [], array $discriminators = [], array $discriminatorColumns = [], array $inheritanceTypes = [], array $uniques = [], array $overrides = [])
+    {
+        $this->associations = $associations;
+        $this->indexes = $indexes;
+        $this->uniques = $uniques;
+        $this->discriminatorColumns = $discriminatorColumns;
+        $this->discriminators = $discriminators;
+        $this->inheritanceTypes = $inheritanceTypes;
+        $this->overrides = $overrides;
+    }
+
+    public function getSubscribedEvents(): array
+    {
+        return [
+            'loadClassMetadata',
+        ];
+    }
+
+    public function addAssociation(string $class, string $field, array $options): void
```

Comment:
We should remove arrays for this class and use ValueObjects instead. This would solve some problems with named array keys.

Reply:
Is it possible? How do you plan to move this? use aliases as in core split or just release new majors?

Reply:
Sure, this should be possible.

We don't have any BC breaks by moving this component to this bundle, unless we don't use aliases or anything else. We can also run both services in parallel without any problems.